azurerm/internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -331,6 +332,71 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
 				Optional: true,
 			},
 
+			"maintenance_config": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"maintenance_allowed": {
```

Comment:
and maintaince is redundant
```suggestion
						"allowed": {
```